### ISSUE TITLE
Enhancement: can't custom db string -> enum property by IPropertyHandler<string, TEnum>

### ISSUE THREAD

**Issue by fredliex** [mikependon/RepoDB#646]
I define `DbValueAttribute` to assign mapping value in database, ex:
```
public enum SpeedEnum {
   [DbValue("(FX)")] High,     // database field value is (FX)
   [DbValue("(FY)")] Middle,  // database field value is (FY)
   [DbValue("(FZ)")] Low       // database field value is (FZ)
}
```
have generic PropertyHandler
```
public class DbValueEnumPropertyHandler<TEnum>: IPropertyHandler<string, TEnum?> {
   public TEnum? Get(string input, ClassProperty property) { .... }
   public TDbValue Get(TEnum? input, ClassProperty property) { .... }
}
```
unittest:
```
Assert.Equal(Connection.ExecuteQuery<string>($"select @a", new { a = SpeedEnum.Middle }).First(), "(FY)");    // success
Assert.Equal(Connection.ExecuteQuery<SpeedEnum>($"select N'(FY)'").First(), SpeedEnum.Middle);  // fail
```
exception is
```
System.InvalidOperationException
  HResult=0x80131509
  Message=Compiler.DataReader.IsDbNull.TrueExpression: Failed to convert the value expression for property handler ''. Context :: TargetType: SpeedEnum
  Source=<無法評估例外狀況來源>
  StackTrace: 
<無法評估例外狀況堆疊追蹤>

  此例外狀況原先在此呼叫堆疊擲回: 
    [外部程式碼]

內部例外狀況 1:
InvalidOperationException: 類型 'SpeedEnum' 與 'System.String' 間未定義強制型轉運算子。
```
seem `GetClassPropertyParameterInfoIsDbNullFalseValueExpression` in `RepoDb.Core\RepoDb\Reflection\Compiler\Compiler.cs`
call `ConvertExpressionToEnumExpression` when from string to enum
bypass PropertyHandler

hope you can help,
thank you.

**Comment by mikependon** [comment]
We will investigate this one and hopefully to include on the next beta release. When do you need the fix for this?

**Comment by fredliex** [comment]
If possible, I hope within two month
thank you very much

**Comment by mikependon** [comment]
Yeah, that should be possible. Thanks

**Comment by mikependon** [comment]
@fredliex, we have assessed this enhancement request internally.

Firstly, the `DbValueAttribute` is not a standard attribute that can be found from the `System.ComponentModel.DataAnnotations.Schema` namespace.

Secondly, the proposal is an edge-case, which most likely be used to only on few scenarios (very low), but the possible changes to the compiler is huge and would impact to the AOT compilation performance of the enumeration (a bit).

It is unfortunate, but we have decided not to proceed with this.
